cardinal/t_gameSystems_test.go

```diff
@@ -386,6 +386,66 @@ func TestInsultMeat_Failure(t *testing.T) {
 
 // #endregion ProcessCommandTokens Test
 
+// #region Tokeniser System Test
+func TestFishTokens_Success(t *testing.T) {
+	tf := testutils.NewTestFixture(t, nil)
+	MustInitWorld(tf.World)
+	setup()
+
+	var tokens = []string{"look", "WITH", "BOTTLE", "AT", "THE", "window"}
+	const expectedVrb = "Look"
+	const expectedDObj = "Bottle"
+	const expectedIObj = "Window"
+
+	tf.DoTick()
+
+	data := system.NewTokeniserSystem().FishTokens(tokens)
+
+	assert.Equal(t, expectedVrb, data.Verb.String())
+	assert.Equal(t, expectedDObj, data.DirectObject.String())
+	assert.Equal(t, expectedIObj, data.IndirectObject.String())
+}
+
+func TestFishTokens_NoDObj(t *testing.T) {
```

Comment:
this is bug, [burn, at, the, window] should return VRB:Burn, DOBJ:Window, IOBJ:None

Reply:
Actually no, cause there is no direct object. It will be as you say if the tokens were {burn , the, window]. In the test the window is the Indirect object also, I changed the expectedOutput of the DObj to be Window and the IObj to be None but the test and debug result showed error. So the way it was setup first is fine